src/scripts/DataLayerItem.js

```diff
@@ -28,6 +27,94 @@ const itemType = {
   LISTENER_OFF: 'listenerOff'
 };
 
+/**
+ * Constraints for each type of the item configuration.
+ */
+const constraints = {
+  data: {
+    properties: [
+      {
```

Comment:
we could perhaps simplify this by structuring the properties by key, and might make sense to flip `mandatory` to `optional` to match our documentation:

```javascript
event: {
  properties: {
    event: {
      type: 'string',
      optional: false
    },
    info: {
      type: 'object',
      optional: false
    },
    data: {
      type: 'object'
      optional: true
    }
  }
}
```